src/hipBin_util.h

```diff
@@ -206,17 +209,24 @@ string HipBinUtil::getSelfPath() const {
     path = string(wide.begin(), wide.end());
   #else
     char buff[MAX_PATH_CHAR];
-    ssize_t len = ::readlink("/proc/self/exe", buff, sizeof(buff) - 1);
-    if (len > 0) {
-      buff[len] = '\0';
-      path = string(buff);
-      fs::path exePath(path);
-      path = exePath.parent_path().string();
-    } else {
-      cout << "readlink: Error reading the exe path" << endl;
-      perror("readlink");
-      exit(-1);
-    }
+    #if defined(__APPLE__) || defined(__MACOSX)
+      uint32_t len = sizeof(buff) - 1;
+      if(_NSGetExecutablePath(buff, &len) || len <= 0) {
+        cout << "hipcc: Error reading the exe path" << endl;
+        exit(-1);
+      }
+    #else
+      ssize_t len = ::readlink("/proc/self/exe", buff, sizeof(buff) - 1);
+      if (len <= 0) {
+        cout << "hipcc: Error reading the exe path" << endl;
+        perror("readlink");
```

Comment:
This error is very confusing. Initially I thought some tool invoked the `readlink` program, ideas?